engine-nextgen/src/test/kotlin/graphql/nadel/validation/NadelHydrationValidationTest.kt

```diff
@@ -724,6 +724,125 @@ class NadelHydrationValidationTest : DescribeSpec({
             assert(error.duplicates.map { it.name }.toSet() == setOf("id"))
         }
 
+        it("fails if hydration field has missing non-nullable arguments with underlying top level fields") {
+            val fixture = NadelValidationTestFixture(
+                    overallSchema = mapOf(
+                            "issues" to """
+                        type Query {
+                            issue: Issue
+                        }
+                        type Issue {
+                            id: ID!
+                        }
+                    """.trimIndent(),
+                            "users" to """
+                        type Query {
+                            user(id: ID!, other: Boolean!): User
+                        }
+                        type User {
+                            id: ID!
+                            name: String!
+                        }
+                        extend type Issue {
+                            creator(someArg: ID!, other: Boolean): User @hydrated(
+                                service: "users"
+                                field: "user"
+                                arguments: [
+                                    {name: "id", value: "$argument.creator"}
+                                ]
+                            )
+                        }
+                    """.trimIndent(),
+                    ),
+                    underlyingSchema = mapOf(
+                            "issues" to """
+                        type Query {
+                            issue: Issue
+                        }
+                        type Issue {
+                            id: ID!
+                            creator: ID!
+                        }
+                    """.trimIndent(),
+                            "users" to """
+                        type Query {
+                            user(id: ID!, other: Boolean!): User
+                        }
+                        type User {
+                            id: ID!
+                            name: String!
+                        }
+                    """.trimIndent(),
+                    ),
+            )
+
+            val errors = validate(fixture)
+            assert(errors.map { it.message }.isNotEmpty())
+
+            val error = errors.assertSingleOfType<MissingHydrationActorFieldArgument>()
```

Comment:
Just occurred to me, you need to create a new class for this.

I think we rename MissingHydrationActorFieldArgument → NonExistentHydrationActorFieldArgument
And create a new MissingHydrationActorFieldArgument

Because the current impl will generate the error message

```
Hydration on field $of references non-existent argument $argument on hydration actor $s.$af
```